components/equihash/src/verify.rs

```diff
@@ -163,7 +206,12 @@ fn expand_array(vin: &[u8], bit_len: usize, byte_pad: usize) -> Vec<u8> {
     vout
 }
 
-fn indices_from_minimal(minimal: &[u8], c_bit_len: usize) -> Vec<u32> {
+fn indices_from_minimal(p: Params, minimal: &[u8]) -> Result<Vec<u32>, Error> {
+    let c_bit_len = p.collision_bit_length();
+    if minimal.len() != (1 << p.k) * (c_bit_len + 1) / 8 {
```

Comment:
I think this might change the order of operations in ways that affect rounding, I'm not positive, but it may be worth adding parens just to be sure.

```suggestion
    if minimal.len() != (1 << p.k) * ((c_bit_len + 1) / 8) {
```

Reply:
The intended order of operations here is actually:
```suggestion
    if minimal.len() != ((1 << p.k) * (c_bit_len + 1)) / 8 {
```
So yes, I'll add this for clarity.

Reply:
oh shit, does that mean the other one you linked was wrong? :grimacing:

Reply:
No, for two reasons:

- `k >= 3`, so `1 << k` is always divisible by 8.
- The associativity of `*` and `/` in Rust is left to right, matching the order of operations now enforced here.

Reply:
But the old version had the associativity backwards right?

```
if (k >= n) || (n % 8 != 0) || (soln_len != (1 << k) * ((n / (k + 1)) as usize + 1) / 8)
```

so this was requiring that `n / (k+1) + 1` is divisible by 8.